src/Approximations/overapproximate.jl

```diff
@@ -597,3 +598,61 @@ function blocks_linear_map(cpa::CartesianProductArray{N, T},
 
     return MinkowskiSumArray(array)
 end
+
+"""
+    overapproximate(cap::Intersection{N,
+                            <:CartesianProductArray{N},
+                            <:AbstractPolyhedron{N}},
+                       ::Type{CartesianProductArray}, oa) where {N}
+
+Return the intersection of the cartesian product of a finite number of convex sets and a polyhedron.
+
+### Input
+
+ - `cap` -- Lazy intersection of cartesian product array and polyhedron
+ - `Y`   -- polyhedron
+ - `Type{CartesianProductArray}` -- argument to specify decomposed overapproximation
```

Comment:
```suggestion
 - `CartesianProductArray` -- type for dispatch
```
(This is what we usually write there - don't ask :wink:)